protocol/README.md

```diff
@@ -4,12 +4,11 @@ This document describes the canonical system architecture used to encrypt and de
 
 ## Architecture
 
-![arch](../diagrams/arch.png)
-
-The canonical architecture contains three major components.
+The canonical architecture contains four major components.
 
 * *TDF Client* - Initiates and drives the TDF encryption and decryption workflows. Only component with access to the content (ciphertext or plaintext).
-* *Entity Attribute Service* (EAS) - Responsible for authenticating TDF Clients and vending them signed [Entity Objects](../schema/EntityObject.md), which act as bearer tokens. The source of truth for attribute provenance and association. A single architecture can have many EAS instance running, for instance, for a TDF Client to join attributes across mulitple orginizations.
+* *Virtru Identity Provider (IdP) (Keycloak)* - From Wikipedia:  "Keycloak is an open source software product to allow single sign-on with Identity and Access Management aimed at modern applications and services. As of March 2018 this JBoss community project is under the stewardship of Red Hat who use it as the upstream project for their RH-SSO product."  Keycloak is open source Apache License 2.0 code.
+* *Attribute Provider* (AP) - A web service that will receive requests from Virtru Protocol Mapper containing information about authenticated users and return our custom OIDC/OAuth claims in response.  It is the responsibility of Attribute Provider to transform 3rd party IdP claims/metadata to ABAC attributes.  It returns a Virtru [Claims Object](../schema/ClaimsObject.md).
 * *Key Access Service* (KAS) - Responsible for authorizing and granting TDF Clients access to rewrapped data key material. If authorized, TDF Clients can use this rewrapped data key to decrypt TDF ciphertext.
```

Comment:
```suggestion
* *Key Access Service* (KAS) - Responsible for authorizing and granting TDF Clients access to rewrapped data key material. If authorized, TDF Clients can use this rewrapped data key to decrypt TDF ciphertext. A valid OIDC token with [TDF Claims](../schema/ClaimsObject.md) must be used as a bearer token when communicating with KAS. An otherwise valid and trusted OIDC token without TDF claims will be rejected as a bearer token by KAS.
```